src/reducers/events.js

```diff
@@ -4,15 +4,101 @@ const events_test_data = [{name: "Soccer game", location: "Empire Field, Vancouv
                           {name: "MTG Tournament", location: "Vancouver Convention Centre", date: new Date("2020-01-04"), genre: ['games', 'card games', 'tournaments', 'magic'], creator: {name: "Joan Smith"}, participants: [{name: "Daniel Boyce"}, {name: "Carol Wisnewski"}, {name: "Samuel Zubrus"}, {name: "Gary Lee"}, {name: "Andrew Gergich"}, {name: "Jennifer Quincey"}, {name: "Joel Quincey"}, {name: "Joel Harland"}, {name: "Carla Gregg"}], description: "Magic The Gathering tournament. Winner takes home $25,000 grand prize", comments: []}];
 
 
+
+
+const addParticipant = (user, event, events) => {
+        let j = -1;
+        for (let i = 0; i < events.length; i++) {
+            if (JSON.stringify(events[i]) === JSON.stringify(event)) {
+                j = i;
+            }
+        }
+        let containsUser = false;
+        for (let i = 0; i < event.participants.length; i++) {
+            if (JSON.stringify(event.participants[i]) === JSON.stringify(user)) {
+                containsUser = true;
+            }
+        }
+        if (!containsUser) {
```

Comment:
Same as above for finding user existence. I think findIndex() or participants.include(user) should be better alternatives.

Reply:
I think you also have the same scenario for some of the code below. I won't leave a comment but if you could fix those as well thank you.